### PR TITLE
[MAINT] update precommit configuration

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- only exclude running black on subset of files to make sure other hooks are applied
- fix flake8 and pyugrade errors that this revelead

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@bthirion @man-shu 

this one feels fairly uncontroversial

will merge this tomorrow unless I hear otherwise from you

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -66,6 +52,27 @@ repos:
...
     -   id: black
         args: [--config, pyproject.toml]
+        exclude: |
+            (?x)^(
+                nilearn/datasets/tests/data/localizer_behavioural.tsv
+                | nilearn/externals/.*
+                | nilearn/plotting/data/js/.*js
+                | nilearn/_version.py
+                | examples/01_plotting/plot_3d_map_to_surface_projection.py
+                | examples/01_plotting/plot_surf_atlas.py
```
Comment (man-shu) [inline .pre-commit-config.yaml:62]: I was just wondering why we don't want to blackify all these files.

**Review by man-shu — APPROVED** [review]
Otherwise looks good to me!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Remi-Gau) [reply .pre-commit-config.yaml:62]: Long story short, using black was implemented "module by module" and the files left there are those that are still involved in some pull requests.

Several of those can be "blackified" once we have made progress on the surface object front.

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**